test/integration/subprocess_TEST.cc

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <gtest/gtest.h>
+
+#include <gz/utils/Subprocess.hh>
+
+#include <chrono>
+#include <thread>
+
+using Subprocess = gz::utils::Subprocess;
+
+static constexpr const char * kExecutablePath = SUBPROCESS_EXECUTABLE_PATH;
+
+/////////////////////////////////////////////////
+TEST(Subprocess, CreateInvalid)
+{
+  auto proc = Subprocess({"an_executable_that_will_never_exist"});
+  EXPECT_FALSE(proc.Alive());
+}
+
+/////////////////////////////////////////////////
+TEST(Subprocess, CreateValid)
+{
+  auto proc = Subprocess({kExecutablePath, "--help"});
+
+  // Sleep for just a bit to guarantee executable finishes
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
Why you don't use `Join` here ? instead of a waiting 100 ms ?

Reply:
Ah, I was originally going to do some checking of `Alive` here, but ended up not.